yt_dlp/extractor/instagram.py

```diff
@@ -195,18 +195,19 @@ def get_count(keys, kind):
                                 lambda x: x['%ss' % kind]['count'])))
                     if count is not None:
                         return count
+
             like_count = get_count('preview_like', 'like')
             comment_count = get_count(
                 ('preview_comment', 'to_comment', 'to_parent_comment'), 'comment')
 
             comments = [{
-                'author': comment.get('user', {}).get('username'),
-                'author_id': comment.get('user', {}).get('id'),
-                'id': comment.get('id'),
-                'text': comment.get('text'),
-                'timestamp': int_or_none(comment.get('created_at')),
+                'author': comment.get('node', {}).get('owner', {}).get('username'),
```

Comment:
what's the point of a list comprehension if you are going to call `comment.get('node', {})` this multiple times?
Also it's a personal pref but the list comprehension is looking over complicated.

```py
 for comment in media.get('edge_media_to_parent_comment', {}).get('edges', []):
   comment_dict = comment.get('node', {})
   comment_text = comment_dict.get('text')
   if comment_text:
     #Append to comments using the `comment_dict` and reuse `comment_text`
```

Reply:
Done.